x/posts/internal/types/msg_reactions_test.go

```diff
@@ -0,0 +1,98 @@
+package types_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/desmos-labs/desmos/x/posts/internal/types"
+	"github.com/stretchr/testify/require"
+
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+var msgRegisterReaction = types.NewMsgRegisterReaction(testOwner, ":smile:", "https://smile.jpg",
+	"4e188d9c17150037d5199bbdb91ae1eb2a78a15aca04cb35530cccb81494b36e")
+
+func TestMsgRegisterReaction_Route(t *testing.T) {
+	actual := msgRegisterReaction.Route()
+	require.Equal(t, "posts", actual)
+}
+
+func TestMsgRegisterReaction_Type(t *testing.T) {
+	actual := msgRegisterReaction.Type()
+	require.Equal(t, "register_reaction", actual)
+}
+
+func TestMsgRegisterReaction_ValidateBasic(t *testing.T) {
+	tests := []struct {
+		name  string
+		msg   types.MsgRegisterReaction
+		error error
+	}{
+		{
+			name: "Invalid creator returns error",
+			msg: types.NewMsgRegisterReaction(nil, ":smile:", "https://smile.jpg",
+				"4e188d9c17150037d5199bbdb91ae1eb2a78a15aca04cb35530cccb81494b36e"),
+			error: sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, fmt.Sprintf("Invalid creator address: %s", "")),
+		},
+		{
+			name: "Empty short code returns error",
+			msg: types.NewMsgRegisterReaction(testOwner, "", "https://smile.jpg",
+				"4e188d9c17150037d5199bbdb91ae1eb2a78a15aca04cb35530cccb81494b36e"),
+			error: sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "reaction short code must be an emoji short code"),
+		},
+		{
+			name: "Invalid short code returns error",
+			msg: types.NewMsgRegisterReaction(testOwner, ":smile", "https://smile.jpg",
+				"4e188d9c17150037d5199bbdb91ae1eb2a78a15aca04cb35530cccb81494b36e"),
+			error: sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "reaction short code must be an emoji short code"),
+		},
+		{
+			name: "Empty value returns error",
+			msg: types.NewMsgRegisterReaction(testOwner, ":smile:", "",
+				"4e188d9c17150037d5199bbdb91ae1eb2a78a15aca04cb35530cccb81494b36e"),
+			error: sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "reaction value should be a URL or an emoji unicode"),
+		},
+		{
+			name: "Invalid value returns error (url)",
+			msg: types.NewMsgRegisterReaction(testOwner, ":smile:", "htp://smile.jpg",
+				"4e188d9c17150037d5199bbdb91ae1eb2a78a15aca04cb35530cccb81494b36e"),
+			error: sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "reaction value should be a URL or an emoji unicode"),
+		},
+		{
+			name: "Invalid value returns error (unicode)",
+			msg: types.NewMsgRegisterReaction(testOwner, ":smile:", "U+1",
+				"4e188d9c17150037d5199bbdb91ae1eb2a78a15aca04cb35530cccb81494b36e"),
+			error: sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "reaction value should be a URL or an emoji unicode"),
+		},
+		{
+			name: "Invalid subspace returns error",
+			msg: types.NewMsgRegisterReaction(testOwner, ":smile:", "https://smile.jpg",
+				"1234"),
+			error: sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("reaction subspace must be a valid sha-256 hash")),
+		},
+	}
```

Comment:
Can we add the following case as well? I would like to make sure registering an emoji reaction works properly

```golang
{
	name: "Valid emoji value returns no error",
	msg: types.NewMsgRegisterReaction(testOwner, ":smile:", "💙", "4e188d9c17150037d5199bbdb91ae1eb2a78a15aca04cb35530cccb81494b36e"),
}
```

Reply:
In order to make sure this will work we need a converter from emoji to unicode. With the go-emoji utils I can make one in a couple of minutes!

Reply:
I'm actually not sure why we need that. Why is so? As far as I know, emojis can be converted into bytes and stored properly inside the chain store without much problems. This is also shown by the currently stored reactions that have the form of emojis. 

Why do we now need to convert them to unicode explicitly? Aren't they converted when stored in any case?